scripts/process/process.cell_precip.R

```diff
@@ -0,0 +1,53 @@
+#' take the compiled-precip object and turn it into 
+#' a single sp.points.d.f w/ proper classes per timestep
+process.cell_precip <- function(viz){
+  library(sp)
+  deps <- readDepends(viz)
+  checkRequired(deps, c("classified-precip", "clip"))
+  clip <- deps$clip
+  offset <- viz[["cell.offset"]]
+  pts <- deps$`classified-precip`
+  
+  # build a set of square polygons for each centroid
+  pols.out <- list()
+  for (j in seq_len(length(pts))){
+    poly <- getPoly(pts[j,1]@coords[1], pts[j,1]@coords[2], id = paste0('p',j), offset)
+    pols.out <- append(pols.out, poly)
+  }
+  
+  
+  SpP = SpatialPolygons(pols.out, 1:j, proj4string = CRS(proj4string(pts)))
+  sp <- as(SpP, 'SpatialPolygonsDataFrame')
+  sp@data <- pts@data
```

Comment:
This kind of stuff worries me. Why not use the sp::SpatialPolygonsDataFrame creator?

Reply:
What would that look like here?

Reply:
As long as you know the order of how SpP was made which you do from getPoly, something like this:
sp_data <- pts@data
row.names(sp_data) <- spChFIDs(SpP)
sp <- SpatialPolygonsDataFrame(SpP, sp_data, match.ID = TRUE)

Reply:
:+1: